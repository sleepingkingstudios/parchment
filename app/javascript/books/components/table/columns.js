import ResponsiveActions from 'components/responsive-actions';
import { injectProps } from 'utils/react';

const generateColumns = ({ actions, baseUrl, useDestroyRequest }) => ([
  {
    label: 'Title',
    prop: 'title',
    width: 5,
  },
  {
    label: 'Publisher',
    prop: 'publisherName',
    width: 4,
  },
  {
    label: false,
    prop: 'actions',
    value: injectProps(
      ResponsiveActions,
      {
        actions,
        baseUrl,
        resourceName: 'book',
        useDestroyRequest,
      },
    ),
    width: 3,
  },
]);

export default generateColumns;
