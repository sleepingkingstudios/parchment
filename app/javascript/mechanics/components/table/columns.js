const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Description',
    prop: 'shortDescription',
  },
  {
    label: ' ',
    prop: 'actions',
    value: null,
  },
];

export default columns;
