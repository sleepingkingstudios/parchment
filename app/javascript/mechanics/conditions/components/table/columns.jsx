import ConditionsTableActions from './actions';

const columns = [
  {
    label: 'Name',
    prop: 'name',
    width: 3,
  },
  {
    label: 'Description',
    prop: 'shortDescription',
    width: 6,
  },
  {
    label: false,
    prop: 'actions',
    value: ConditionsTableActions,
    width: 3,
  },
];

export default columns;
