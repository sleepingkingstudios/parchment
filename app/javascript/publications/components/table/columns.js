import PublicationsTableActions from './actions';

const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Publisher',
    prop: 'publisherName',
  },
  {
    label: 'Date',
    prop: 'publicationDate',
  },
  {
    label: 'Playtest',
    prop: 'playtest',
  },
  {
    label: ' ',
    prop: 'actions',
    value: PublicationsTableActions,
  },
];

export default columns;
