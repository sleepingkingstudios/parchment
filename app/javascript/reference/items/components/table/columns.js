import ResponsiveActions from 'components/responsive-actions';
import { injectProps } from 'utils/react';

const generateColumns = ({ actions, baseUrl, useDestroyRequest }) => ([
  {
    label: 'Name',
    prop: 'name',
    width: 6,
  },
  {
    label: 'Cost',
    prop: 'cost',
    width: 3,
  },
  {
    label: false,
    prop: 'actions',
    value: injectProps(
      ResponsiveActions,
      {
        actions,
        baseUrl,
        resourceName: 'item',
        useDestroyRequest,
      },
    ),
    width: 3,
  },
]);

export default generateColumns;
