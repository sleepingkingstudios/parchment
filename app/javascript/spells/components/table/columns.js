import SpellsTableActions from './actions';

import { exists } from '../../../utils/object';
import {
  capitalize,
  truncate,
} from '../../../utils/string';

const sourceLink = (spell) => {
  const { source } = spell;

  return exists(source) ? source.name : 'Homebrew';
};

const spellDescription = (spell) => {
  if (spell.shortDescription) {
    return truncate(spell.shortDescription, 60);
  }

  return truncate(spell.description, 60);
};

const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Source',
    prop: 'source',
    value: sourceLink,
  },
  {
    label: 'School',
    prop: 'school',
    value: spell => capitalize(spell.school),
  },
  {
    label: 'Level',
    prop: 'level',
  },
  {
    label: 'Description',
    prop: 'description',
    value: spellDescription,
  },
  {
    label: ' ',
    prop: 'actions',
    value: SpellsTableActions,
  },
];

export default columns;
