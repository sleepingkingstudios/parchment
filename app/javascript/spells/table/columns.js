import { capitalize } from '../../utils/string';

const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'School',
    prop: 'school',
    value: spell => capitalize(spell.school),
  },
  {
    label: 'Level',
    prop: 'level',
  },
];

export default columns;
