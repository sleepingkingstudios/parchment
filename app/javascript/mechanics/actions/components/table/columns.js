import ActionsTableActions from './actions';
import { columns as mechanicsColumns } from '../../../components/table';

const columns = mechanicsColumns.slice();

columns[columns.length - 1] = {
  label: ' ',
  prop: 'actions',
  value: ActionsTableActions,
};

export default columns;
