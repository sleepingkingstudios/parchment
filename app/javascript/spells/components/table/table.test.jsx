import React from 'react';
import { shallow } from 'enzyme';

import SpellsTable from './table';
import columns from './columns';

describe('<SpellsTable />', () => {
  const data = { spells: [] };
  const resourceName = 'spells';
  const defaultOptions = { data, resourceName };

  describe('with default options', () => {
    const rendered = shallow(<SpellsTable {...defaultOptions} />);

    it('should be an IndexPageTable', () => {
      expect(rendered).toHaveDisplayName('IndexPageTable');
    });

    it('should set the columns', () => {
      expect(rendered).toHaveProp({ columns });
    });
  });
});
