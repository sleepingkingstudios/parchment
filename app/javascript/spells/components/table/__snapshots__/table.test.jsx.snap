// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellsTable /> with spells: Array with many items should match the snapshot 1`] = `
<Table
  cellProps={
    Object {
      "onDelete": [MockFunction],
    }
  }
  columns={
    Array [
      Object {
        "label": "Name",
        "prop": "name",
      },
      Object {
        "label": "Source",
        "prop": "source",
        "value": [Function],
      },
      Object {
        "label": "School",
        "prop": "school",
        "value": [Function],
      },
      Object {
        "label": "Level",
        "prop": "level",
      },
      Object {
        "label": "Description",
        "prop": "description",
        "value": [Function],
      },
      Object {
        "label": " ",
        "prop": "actions",
        "value": [Function],
      },
    ]
  }
  data={
    Array [
      Object {
        "castingTime": "1 action",
        "description": "You conjure a glowing, spectral orb the size of a closed fist and bearing the likeness of everyone's favorite creature, the Flumph. Until the spell ends, the orb sheds bright light in a 20-foot radius and dim light for an additional 20 feet. The light can be colored as you like. The spell ends if you cast it again or dismiss it as an action.

You can move the orb to anywhere within 30 feet of you as a bonus action. If you move more than 30 feet away from the orb, it will float directly to your location at the beginning of your next turn; if it is unable to reach your location, the spell immediately ends. ",
        "duration": "1 hour",
        "id": "823dbe98-ef01-4984-b8b2-34c2946091ce",
        "level": 0,
        "materialComponent": "a pinch of flumph dust",
        "name": "Flumph Lantern",
        "range": "30 feet",
        "ritual": false,
        "school": "evocation",
        "shortDescription": "Floating light illuminates 20 feet + 20 feet dim light",
        "slug": "flumph-lantern",
        "somaticComponent": false,
        "verbalComponent": true,
      },
      Object {
        "castingTime": "1 action",
        "description": "For the duration, you sense the presence of Flumphs within 30 feet of you. The spell can penetrate most barriers, but it is blocked by 1 foot of stone, 1 inch of common metal, a thin sheet of lead, or 3 feet of wood or dirt. ",
        "duration": "Concentration, up to 10 minutes",
        "id": "a1ab75a4-83c8-47d2-ada2-8a9bd70341ae",
        "level": 1,
        "materialComponent": "",
        "name": "Detect Flumph",
        "range": "Self",
        "ritual": true,
        "school": "divination",
        "shortDescription": "Sense Flumphs within 30 feet",
        "slug": "detect-flumph",
        "somaticComponent": true,
        "verbalComponent": true,
      },
      Object {
        "castingTime": "1 bonus action",
        "description": "The next time you hit a creature with a weapon attack before this spell ends, the weapon is wrapped with a ghostly Flumph tentacle as you strike. The attack deals an extra 2d6 radiant damage to the target, which becomes visible if it's invisible, and the target sheds dim light in a 5-foot radius and can't become invisible until the spell ends.

**At Higher Levels.** When you cast this spell using a spell slot of 3rd level or higher, the extra damage increases by 1d6 for each slot level above 2nd. ",
        "duration": "Concentration, up to 1 minute",
        "id": "161c7a6a-54e0-4082-9e9e-a54950b31318",
        "level": 2,
        "materialComponent": "",
        "name": "Spectral Tentacle",
        "range": "Self",
        "ritual": false,
        "school": "evocation",
        "shortDescription": "Next weapon attack deals +2d6 radiant damage",
        "slug": "spectral-tentacle",
        "somaticComponent": false,
        "verbalComponent": true,
      },
      Object {
        "castingTime": "1 minute",
        "description": "You create a 10-foot-radius, 20-foot-tall cylinder of magical energy centered on a point on the ground that you can see within range. Glowing runes appear wherever the cylinder intersects with the floor or other surface. The circle affects all Flumphs in the following ways:

- A Flumph can't willingly enter the cylinder by nonmagical means. If the creature tries to use teleportation or interplanar travel to do so, it must first succeed on a Charisma saving throw. - The creature has disadvantage on attack rolls against targets within the cylinder. - Targets within the cylinder can't be charmed, frightened, or possessed by the creature.

When you cast this spell, you can elect to cause its magic to operate in the reverse direction, preventing a creature of the specified type from leaving the cylinder and protecting targets outside it.

**At Higher Levels:** When you cast this spell using a spell slot of 4th level or higher, the duration increases by 1 hour for each slot level above 3rd. ",
        "duration": "1 hour",
        "id": "07e1e012-7f86-4534-8b02-cf79d2824a9d",
        "level": 3,
        "materialComponent": "a vial of rare salt worth at least 100 gp, which the spell consumes",
        "name": "Magic Circle Against Flumphs",
        "range": "10 feet",
        "ritual": false,
        "school": "abjuration",
        "shortDescription": "Protect 10 ft circle from Flumphs.",
        "slug": "magic-circle-against-flumphs",
        "somaticComponent": true,
        "verbalComponent": true,
      },
    ]
  }
  message="There are no spells matching the criteria."
/>
`;

exports[`<SpellsTable /> with spells: empty Array should match the snapshot 1`] = `
<Table
  cellProps={
    Object {
      "onDelete": [MockFunction],
    }
  }
  columns={
    Array [
      Object {
        "label": "Name",
        "prop": "name",
      },
      Object {
        "label": "Source",
        "prop": "source",
        "value": [Function],
      },
      Object {
        "label": "School",
        "prop": "school",
        "value": [Function],
      },
      Object {
        "label": "Level",
        "prop": "level",
      },
      Object {
        "label": "Description",
        "prop": "description",
        "value": [Function],
      },
      Object {
        "label": " ",
        "prop": "actions",
        "value": [Function],
      },
    ]
  }
  data={Array []}
  message="There are no spells matching the criteria."
/>
`;
